In [1]:
import pandas as pd
import os
import numpy as np
import matplotlib.pyplot as plt
import plotly
plotly.tools.set_credentials_file(username='paranyaj',api_key='RYp3Uubq9Tgbeogsz3jF')



path = '/Users/BOWPJ/Desktop/NVTC'
os.chdir(path)


# Data from: https://catalog.data.gov/dataset/rural-veterans-by-state-e6abb
rural_urban = pd.read_excel('Rural_Veterans_by_State_2014.xlsx')

rural_urban = rural_urban[2:]
rural_urban = rural_urban.reset_index()

rural_urban.columns = ['state','vetpop_rural','vetpop_urban','perc_f_rural','perc_f_urban','unemp_vet_rural','unemp_vet_urban','bel_pov_inc_rural','bel_pov_inc_urban',
                       'disabil_rate_rural','disabil_rate_urban','med_inc_rural','med_inc_urban',
                       '18-34_rural','18-34_urban','35-54_rural','35-54_urban','55-64_rural','55-64_urban',
                       '65-74_rural','65-74_urban','75+_rural','75+_urban','gulf2_rural','gulf2_urban',
                       'gulf1_rural','gulf1_urban','vietnam_rural','vietnam_urban','korean_rural','korean_urban',
                       'world2_rural','world2_urban','less_highschool_rural','less_highschool_urban','highschool_rural',
                       'highschool_urban','some_college_rural','some_college_urban','bachelor_rural','bachelor_urban']
                      

rural_urban = rural_urban.replace('X',np.nan)
rural_urban = rural_urban[:52]


In [2]:
rural_urban.head()

state  vetpop_rural  vetpop_urban  perc_f_rural  perc_f_urban  \
0     Alabama      142884.0      201420.0           7.6           9.0   
1      Alaska       21470.0       48378.0           8.4          15.0   
2     Arizona       62905.0      420121.0           7.0           8.3   
3    Arkansas      101788.0      106098.0           6.3           9.6   
4  California      142266.0     1551336.0           5.4           7.7   

   unemp_vet_rural  unemp_vet_urban  bel_pov_inc_rural  bel_pov_inc_urban  \
0              4.0              6.5                8.5                8.8   
1             12.3              6.4                4.9                4.0   
2              5.3              7.0                8.4                7.4   
3              4.2              7.6                9.1                8.1   
4              8.6              7.8                8.8                7.6   

   disabil_rate_rural       ...        world2_rural  world2_urban  \
0                36.0       ...                 3.4           4.7   
1                30.5       ...                 1.3           2.9   
2                34.4       ...                 4.1           5.8   
3                38.7       ...                 5.2           4.2   
4                32.9       ...                 6.4           6.7   

   less_highschool_rural  less_highschool_urban  highschool_rural  \
0                   10.1                    5.3              35.2   
1                    4.1                    3.8              31.8   
2                    6.2                    5.2              23.2   
3                   11.1                    7.2              38.6   
4                    5.9                    5.8              24.3   

   highschool_urban  some_college_rural  some_college_urban  bachelor_rural  \
0              24.6                37.1                39.0            17.6   
1              19.4                38.0                49.4            26.2   
2              22.1                43.2                42.8            27.4   
3              28.7                36.1                37.2            14.2   
4              19.5                42.5                42.0            27.3   

   bachelor_urban  
0            31.1  
1            27.4  
2            29.9  
3            26.9  
4            32.7  

[5 rows x 41 columns]

# Characteristics (%) of veterans in each state

In [3]:
# 2014 data from: https://www.census.gov/library/visualizations/2015/comm/veterans-statistics.html
characteristics = pd.read_excel('vet-state-characteristics.xlsx')
characteristics = characteristics.dropna(axis=1,thresh=20)
characteristics.columns = ['state','total_vet_pop','female','med_inc','gulf2','gulf1','vietnam','korean','world2','peace','bachelor','enrolled',
                          'uninsured','disabil','VA','18-29','30-44','45-54','55-64','65+','-10k','10k-14.9k','15k-24.9k','25k-34.9k',
                          '35k-49.9k','50k-74.9','75k-99.9k','100k-149.9k','150k-199.9k','200k+']

characteristics = characteristics.iloc[2:55]
characteristics = characteristics.iloc[1:]
characteristics.index = range(0,len(characteristics))
characteristics.head()


state total_vet_pop   female med_inc gulf2 gulf1 vietnam korean  \
0     Alabama        388865   7.6911   57935  13.4  14.6    33.6    8.9   
1      Alaska         71004  12.7106   85518  24.2  17.8    30.8    4.2   
2     Arizona        522382  7.69207   60197  11.5  12.1    33.6     11   
3    Arkansas        237311  7.13915   50669  11.8  12.8    34.5    9.7   
4  California       1893539  6.85257   73722  11.4  10.7    33.9   10.8   

  world2 peace  ...   -10k 10k-14.9k 15k-24.9k 25k-34.9k 35k-49.9k 50k-74.9  \
0    5.7  23.8  ...    3.5       3.6       8.9      10.7      15.8       21   
1    2.5  20.5  ...    1.7       2.4       4.6       5.8      10.1     17.7   
2    7.3  24.5  ...   3.36      3.03       8.2     10.07     15.44    21.44   
3    6.2    25  ...   4.02      4.36     11.12     12.29     17.43     22.2   
4    8.5  24.7  ...   2.61      2.94      6.61      7.91      12.1    18.69   

  75k-99.9k 100k-149.9k 150k-199.9k 200k+  
0      14.3          14         4.7   3.6  
1      16.8        22.8          11   7.1  
2     14.47          15        5.15  3.83  
3     12.55       10.94        2.93  2.16  
4     14.58       18.64         8.3  7.61  

[5 rows x 30 columns]

In [4]:
suicide_rate = pd.read_excel('suicide_rate.xlsx')
suicide_rate.head()

state  total_vet  total_all  18-34_vet  18-34_all  35-54_vet  \
0     Alabama       37.8       18.8       86.1       16.1       45.7   
1      Alaska       54.4       30.0      101.7       41.2       56.7   
2     Arizona       53.6       23.7       76.8       19.7       74.5   
3    Arkansas       48.1       22.0      116.4       20.1       50.9   
4  California       39.1       14.0       47.1       10.0       45.0   

   35-54_all  55-74_vet  55-74_all  75+_vet  75+_all  
0       22.3       29.3       18.0     28.7     16.7  
1       28.4        NaN        NaN      NaN      NaN  
2       24.9       42.4       25.9     48.3     25.9  
3       25.6       40.5       21.6     35.4     15.8  
4       14.3       32.3       16.5     42.3     21.6

In [5]:
import plotly.plotly as py
import plotly.graph_objs as go
import plotly
plotly.tools.set_credentials_file(username='paranyaj',api_key='RYp3Uubq9Tgbeogsz3jF')
gender_suicide = pd.read_excel('gender_vet_suicide.xlsx')

male = go.Scatter(x=gender_suicide['Year'],y=gender_suicide['Male'],mode='lines+markers',name='Male',
                 line=dict(color='blue'))
female = go.Scatter(x=gender_suicide['Year'],y=gender_suicide['Female'],
                    mode='lines+markers',name='Female',line=dict(
    color='rgba(231, 84,128 , .8)'))
total = go.Scatter(x=gender_suicide['Year'],y=gender_suicide['Total'],mode='lines+markers',name='Total',
                  line=dict(color='green'))
data = [male,female,total]
layout = dict(title='Veteran Suicide Rate From 2001 to 2014',
             yaxis=dict(title='Suicide Rate'),
             xaxis = dict(title='Year'))
fig = dict(data=data,layout=layout)
py.iplot(fig,file='gender_suicide')


## Characteristics + crude suicide rate for each state

In [6]:
import plotly.figure_factory as ff
# Adding suicide rate to characteristics data frame
characteristics['suicide_rate'] = suicide_rate['total_vet']

### Correlation between each characteristic and crude suicide rate

In [7]:
cols = characteristics.columns.drop('state')
characteristics[cols] = characteristics[cols].apply(pd.to_numeric, errors='coerce')
characteristics_nostate = characteristics.drop('state',axis=1)
characteristics_nostate.corr()['suicide_rate'].sort_values(ascending=False)

suicide_rate     1.000000
uninsured        0.572600
50k-74.9         0.516602
55-64            0.516001
vietnam          0.460681
VA               0.395031
gulf1            0.363383
disabil          0.357463
30-44            0.332506
35k-49.9k        0.265868
18-29            0.251435
75k-99.9k        0.194078
female           0.153577
gulf2            0.148778
enrolled         0.148345
25k-34.9k        0.144697
45-54            0.055993
15k-24.9k        0.016018
10k-14.9k        0.015760
-10k            -0.057223
100k-149.9k     -0.163146
bachelor        -0.205426
med_inc         -0.208567
total_vet_pop   -0.266475
150k-199.9k     -0.276720
peace           -0.285262
world2          -0.355355
65+             -0.365545
200k+           -0.384025
korean          -0.407222
Name: suicide_rate, dtype: float64

In [8]:
characteristics_nostate.corr()['uninsured']

total_vet_pop    0.094695
female           0.301224
med_inc         -0.315853
gulf2            0.219770
gulf1            0.681035
vietnam          0.319735
korean          -0.559461
world2          -0.590325
peace           -0.294594
bachelor        -0.335424
enrolled         0.238474
uninsured        1.000000
disabil          0.409497
VA               0.300534
18-29            0.277839
30-44            0.566553
45-54            0.320760
55-64            0.515581
65+             -0.573335
-10k             0.117184
10k-14.9k        0.137605
15k-24.9k        0.148906
25k-34.9k        0.287191
35k-49.9k        0.339558
50k-74.9         0.391013
75k-99.9k        0.020932
100k-149.9k     -0.276846
150k-199.9k     -0.310102
200k+           -0.397760
suicide_rate     0.572600
Name: uninsured, dtype: float64

## Plotting Uninsured vs. Suicide Rate

In [9]:
%matplotlib inline
import pylab

uninsured_scatter = go.Scatter(x=characteristics['uninsured'],
          y = characteristics['suicide_rate'],
                               text = characteristics['state'],
          mode = 'markers',
          marker = dict(size=10,color='blue',line=dict(width=2,color='black')))


layout = dict(title='Percentage of Uninsured Veterans vs. Suicide Rate',
             yaxis=dict(title='Crude Suicide Rate'),
             xaxis = dict(title='Percentage of Uninsured Veterans'))
fig = dict(data=[uninsured_scatter],layout=layout)
py.iplot(fig, filename='uninsured')


High five! You successfully sent some data to your account on plotly. View your plot in your browser at https://plot.ly/~paranyaj/0 or inside your plot.ly account where it is named 'uninsured'


In [10]:
uninsured_rural_scatter = go.Scatter(x=characteristics['uninsured'],
          y = characteristics['suicide_rate'],
                               text = characteristics['state'],
          mode = 'markers',
          marker = dict(size=10,color=characteristics['disabil'],colorscale='Viridis',showscale=True,line=dict(width=2,color='black')))


layout = dict(title='Percentage of Veterans Uninsured and Disability Rate',
             yaxis=dict(title='Crude Suicide Rate'),
             xaxis = dict(title='Percentage of Uninsured Veterans'))
fig = dict(data=[uninsured_rural_scatter],layout=layout)
py.iplot(fig, filename='uninsured_rural')

In [11]:
disabil_uninsured = go.Scatter(x=characteristics['uninsured'],
          y = characteristics['disabil'],
                               text = characteristics['state'],
          mode = 'markers',
          marker = dict(size=10,color='blue',line=dict(width=2,color='black')))


layout = dict(title='Percentage of Uninsured Veterans vs. Percentage of Veterans with Disability',
             yaxis=dict(title='Percentage of Veterans with Disability'),
             xaxis = dict(title='Percentage of Uninsured Veterans'))
fig = dict(data=[disabil_uninsured],layout=layout)
py.iplot(fig, filename='disabil_uninsured')

Notice: percentage of uninsured veterans is most highly correlated with suicide rate.

In [12]:
income_va = go.Scatter(x=characteristics['med_inc'],
          y = characteristics['VA'],
                               text = characteristics['state'],
          mode = 'markers',
          marker = dict(size=10,color='black',line=dict(width=2,color='black')))

layout = dict(title='Median Income vs. VA Health Care',
             yaxis=dict(title='Veterans Using VA Health Care (%)'),
             xaxis = dict(title='Median Income ($)'))

fig = dict(data=[income_va],layout=layout)
py.iplot(fig, filename='medianincome_va')

### Do states with more percentage of veterans in rural areas have more suicides?

In [13]:
cols = characteristics.columns.drop('state')
characteristics[cols] = characteristics[cols].apply(pd.to_numeric, errors='coerce')

rate_rural_urban = pd.DataFrame({'state':rural_urban['state'],'suicide_rate':suicide_rate['total_vet']})
rate_rural_urban['percent_rural'] = rural_urban['vetpop_rural']/characteristics['total_vet_pop']
rate_rural_urban['percent_urban'] = rural_urban['vetpop_urban']/characteristics['total_vet_pop']
rate_rural_urban[['suicide_rate','percent_rural','percent_urban']].corr()
rate_rural_urban.corr()['suicide_rate']



suicide_rate     1.000000
percent_rural    0.431181
percent_urban   -0.359680
Name: suicide_rate, dtype: float64

In [14]:
rural_uninsured = rate_rural_urban[['state','percent_rural']].merge(characteristics[['state','uninsured']])
rural_uninsured.corr()

percent_rural  uninsured
percent_rural       1.000000   0.330259
uninsured           0.330259   1.000000

### Which states have the highest percentage of veterans in rural areas?

In [15]:
# Reminder than suicide_rate is crude rate (per 100,000 people)
rate_rural_urban.sort_values(by='percent_rural',ascending=False).head(10)

state  suicide_rate  percent_rural  percent_urban
19          Maine          48.3       0.600006       0.298419
45        Vermont          54.9       0.573613       0.328752
48  West Virginia          49.0       0.457315       0.425449
26        Montana          68.6       0.433191       0.461855
24    Mississippi          42.4       0.429210       0.440039
3        Arkansas          48.1       0.428922       0.447084
41   South Dakota          47.9       0.423755       0.499043
34   North Dakota          45.8       0.391839       0.552251
29  New Hampshire          37.2       0.390425       0.492207
17       Kentucky          48.6       0.375279       0.521204

In [16]:
most_rural=rate_rural_urban.sort_values(by='percent_rural',ascending=False).iloc[:20]
most_rural_states = most_rural.state.tolist()
characteristics[characteristics['state'].isin(most_rural_states)]



state  total_vet_pop     female  med_inc  gulf2  gulf1  vietnam  \
0          Alabama         388865   7.691101    57935   13.4   14.6     33.6   
1           Alaska          71004  12.710552    85518   24.2   17.8     30.8   
3         Arkansas         237311   7.139155    50669   11.8   12.8     34.5   
15            Iowa         226175   5.791533    55514    9.7   10.2     34.1   
17        Kentucky         312365   6.642550    53304   11.1   13.0     34.6   
19           Maine         126842   6.382744    55814    8.6   11.5     35.3   
24     Mississippi         200748   7.738558    52573   13.8   13.7     32.6   
25        Missouri         479828   6.327684    54311    9.8   11.7     34.5   
26         Montana          94404   6.857760    52195   10.7   12.1     35.9   
29   New Hampshire         112790   7.363241    69664    8.3   10.4     35.2   
33  North Carolina         724295   8.752925    59220   13.0   14.7     33.0   
34    North Dakota          53157   7.583197    61793   16.0   11.0     33.7   
36        Oklahoma         312492   7.082421    55620   13.0   13.1     34.9   
40  South Carolina         391660   8.518102    58792   12.8   14.8     34.0   
41    South Dakota          67886   7.777745    56847   13.1   11.7     32.5   
42       Tennessee         484901   7.274268    55244   11.5   13.0     34.8   
45         Vermont          48456   6.599802    58413    7.4    8.6     35.8   
48   West Virginia         159448   5.175982    48786    8.8   11.2     36.4   
49       Wisconsin         408870   6.353120    56311    8.6   10.1     34.0   
50         Wyoming          50852   7.401872    62865   14.4   15.0     34.3   

    korean  world2  peace      ...       10k-14.9k  15k-24.9k  25k-34.9k  \
0      8.9     5.7   23.8      ...            3.60       8.90      10.70   
1      4.2     2.5   20.5      ...            2.40       4.60       5.80   
3      9.7     6.2   25.0      ...            4.36      11.12      12.29   
15    12.8     9.3   23.9      ...            3.33       8.97      11.70   
17     9.3     6.3   25.6      ...            4.21      10.03      11.28   
19    10.4     8.1   26.1      ...            3.85       9.84      11.46   
24     9.4     6.1   24.5      ...            4.41      10.53      11.54   
25    10.7     7.6   25.6      ...            3.60       9.46      11.67   
26    10.1     6.4   24.8      ...            4.28      11.22      12.05   
29    10.4     7.7   28.0      ...            2.17       7.33       8.80   
33     8.3     5.9   25.1      ...            3.23       8.49      10.24   
34    10.4     7.0   21.9      ...            2.69       7.54      10.64   
36     9.5     6.3   23.2      ...            3.97       9.31      11.07   
40     8.4     5.3   24.7      ...            3.32       8.65      10.18   
41    11.8     7.2   23.7      ...            4.07       9.10      10.13   
42     8.9     5.9   25.8      ...            3.77       9.58      11.11   
45    10.7     8.1   29.4      ...            3.46       9.07       9.96   
48    10.6     7.1   25.9      ...            4.27      11.91      13.56   
49    11.2     8.8   27.3      ...            3.11       9.26      11.23   
50     9.4     5.7   21.2      ...            3.61       7.77       9.15   

    35k-49.9k  50k-74.9  75k-99.9k  100k-149.9k  150k-199.9k  200k+  \
0       15.80     21.00      14.30        14.00         4.70   3.60   
1       10.10     17.70      16.80        22.80        11.00   7.10   
3       17.43     22.20      12.55        10.94         2.93   2.16   
15      17.10     23.20      14.68        12.32         3.29   2.59   
17      16.87     21.59      13.72        12.20         3.16   2.91   
19      16.27     22.78      14.99        12.39         3.23   2.32   
24      16.50     21.22      13.34        11.93         3.61   2.65   
25      17.03     22.25      13.66        12.67         3.48   2.64   
26      16.06     22.29      13.66        11.52         2.90   2.28   
29      14.10     19.42  

In [17]:
rate_rural_urban.sort_values(by='suicide_rate',ascending=False).head(10)

state  suicide_rate  percent_rural  percent_urban
26        Montana          68.6       0.433191       0.461855
44           Utah          62.4       0.099749       0.814072
31     New Mexico          59.8       0.206216       0.674155
28         Nevada          59.8       0.079672       0.857469
50        Wyoming          55.0       0.324392       0.568788
45        Vermont          54.9       0.573613       0.328752
1          Alaska          54.4       0.302377       0.681342
36       Oklahoma          53.8       0.327586       0.557861
2         Arizona          53.6       0.120420       0.804241
48  West Virginia          49.0       0.457315       0.425449

Notice: Montana has the highest suicide rate and one of the highest percentage of veterans in rural areas.
2014 suicides in Montana:
* Total suicides: 58
* Age 18-34: < 10
* Age 35-54: 21
* Age 55-74: 10-20
* Total Male: 50-60
* Total Female: <10

In [22]:
# Without Arizona, Utah, Nevada, New Mexico
rate_rural_urban1 = rate_rural_urban[rate_rural_urban.state != 'Arizona']
rate_rural_urban1 = rate_rural_urban1[rate_rural_urban.state != 'Nevada']
rate_rural_urban1 = rate_rural_urban1[rate_rural_urban.state != 'New Mexico']
rate_rural_urban1 = rate_rural_urban1[rate_rural_urban.state != 'Utah']
rate_rural_urban.corr()

/Users/BOWPJ/anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:3: UserWarning:

Boolean Series key will be reindexed to match DataFrame index.

/Users/BOWPJ/anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:4: UserWarning:

Boolean Series key will be reindexed to match DataFrame index.

/Users/BOWPJ/anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:5: UserWarning:

Boolean Series key will be reindexed to match DataFrame index.



suicide_rate  percent_rural  percent_urban
suicide_rate       1.000000       0.431181      -0.359680
percent_rural      0.431181       1.000000      -0.967655
percent_urban     -0.359680      -0.967655       1.000000

Ranking (highest to lowest) for percentage of veterans living in urban areas.

In [17]:
%matplotlib inline
import pylab

rural_per = go.Scatter(x=rate_rural_urban['percent_rural'],
          y = rate_rural_urban['suicide_rate'],
          mode = 'markers',
                       name = 'Rural',
                       text=rate_rural_urban['state'],
                       
          marker = dict(size=10,color='red',line=dict(width=2,color='black')))
urban_per = go.Scatter(x=rate_rural_urban['percent_urban'],
                      y = rate_rural_urban['suicide_rate'],
                       name = 'Urban',
                       text=rate_rural_urban['state'],
                      mode='markers',
                      marker = dict(size=10,color='yellow',
                                   line=dict(width=2,color='black')))
data_area = [rural_per,urban_per]

layout = dict(title='Percentage of Veterans in Rural and Urban Areas vs. Suicide Rate',
             yaxis=dict(title='Suicide Rate'),
             xaxis = dict(title='Percentage of Veterans'))
fig = dict(data=data_area,layout=layout)
py.iplot(fig, filename='rural_urban')


In [82]:
rural_uninsured = go.Scatter(x=rate_rural_urban['percent_rural'],
          y = characteristics['uninsured'],
          mode = 'markers',
                       name = 'Rural',
                       text=rate_rural_urban['state'],
                       
          marker = dict(size=10,color='red',line=dict(width=2,color='black')))
data_area = [rural_uninsured]

layout = dict(title='Percentage of Veterans in Rural Areas vs. Percentage of Veterans Uninsured',
             yaxis=dict(title='Percentage of Veterans Uninsured'),
             xaxis = dict(title='Percentage of Veterans in Rural Areas'))
fig = dict(data=data_area,layout=layout)
py.iplot(fig, filename='rural_uninsured')

### Looking at other characteristics of veterans in states (excluding rural/urban population)

In [18]:
other_rural_char = rural_urban.drop(['vetpop_urban','vetpop_rural','state'],axis=1)
other_rural_char['suicide_rate'] = suicide_rate['total_all']
other_rural_char.corr()['suicide_rate'].sort_values(ascending=False).head(10)

suicide_rate          1.000000
some_college_urban    0.573722
some_college_rural    0.446220
disabil_rate_rural    0.403708
55-64_rural           0.394350
18-34_urban           0.378058
gulf1_urban           0.334110
35-54_urban           0.332889
55-64_urban           0.331035
perc_f_rural          0.320213
Name: suicide_rate, dtype: float64

Correlation between suicide rate in each state for veterans and characteristics of veterans in each state.

### Compare percentage of veteran suicide to general population

In [19]:
diff_all = []
suicide_rate1 = suicide_rate.dropna(subset=['total_vet','18-34_vet','35-54_vet','55-74_vet','75+_vet'],axis=0)
for index, row in suicide_rate1.iterrows():
    if row['total_all'] > row['total_vet']:
        print(row['state'])
    diff = row['total_vet']-row['total_all']
    diff_all.append(diff)
np.mean(diff_all)


21.957142857142859

On average (across all states), the suicide rates for veterans is 21.25% greater than the general population.

In [20]:
diff_1 = []
for index, row in suicide_rate1.iterrows():
    diff = row['18-34_vet']-row['18-34_all']
    diff_1.append(diff)

diff_2 = []
for index, row in suicide_rate1.iterrows():
    diff = row['35-54_vet']-row['35-54_all']
    diff_2.append(diff)
    
diff_3 = []
for index, row in suicide_rate1.iterrows():
    diff = row['55-74_vet']-row['55-74_all']
    diff_3.append(diff)

diff_4 = []
for index, row in suicide_rate1.iterrows():
    diff = row['75+_vet']-row['75+_all']
    diff_4.append(diff)
    
print(np.mean(diff_1))
print(np.mean(diff_2))
print(np.mean(diff_3))
print(np.mean(diff_4))

62.4914285714
29.9285714286
12.9942857143
14.5914285714


Numbers above are the average difference in suicide rate for each age group across states.

### Comparing Suicide Methods between General/Veterans

In [21]:
# suicide rates 2014
# All: Female: 1.5, Male: 2.6

labels = ['Firearm','Suffocation','Poisoning','Other']
f_number = [2999,3296,2533,833]
m_number = [18335,3513,8874,2391]

colors = ['#01579B','#B3E5FC','#03A9F4','#B3E5FC']
fig = {'data':[{'values':[18335,3513,8874,2391],
               'labels':['Firearm','Suffocation','Poisoning','Other'],
               'domain':{'x':[0,0.48]},'name':'General Male Suicide Methods',
               'hoverinfo':'label+percent+name',
               'hole':0.4,'type':'pie'},
              {'values':[2999,3296,2533,833],'labels':['Firearm','Suffocation','Poisoning','Other'],'text':'Female',
              'textposition':'inside','domain':{'x':[0.52,1]},'name':'General Female Suicide Method','hoverinfo':'label+percent"name',
              'hole':0.4,'type':'pie'}],
      'layout':{'title':'General Suicide Methods','annotations':[
            {'font':{'size':20},
            'showarrow':False,'text':'Male','x':0.18,'y':0.5},
            {'font':{'size':20},'showarrow':False,'text':'Female','x':0.85,'y':0.5
            }
        ]}}

pie1 = go.Pie(labels=labels,values=m_number,textinfo='value',textfont=dict(size=10),
             marker=dict(colors=colors, 
                           line=dict(color='black', width=2)))
py.iplot(fig,filename='General Suicide Methods')


In [22]:

colors = ['#01579B','#B3E5FC','#03A9F4','#B3E5FC']
fig = {'data':[{'values':[67.9,10.2,17.2,4.7],
               'labels':['Firearm','Suffocation','Poisoning','Other'],
               'domain':{'x':[0,0.48]},'name':'Veteran Male Suicide Methods',
               'hoverinfo':'label+percent+name',
               'hole':0.4,'type':'pie'},
              {'values':[40.5,32.3,20.2,7],'labels':['Firearm','Suffocation','Poisoning','Other'],'text':'Female',
              'textposition':'inside','domain':{'x':[0.52,1]},'name':'Veteran Female Suicide Method','hoverinfo':'label+percent"name',
              'hole':0.4,'type':'pie'}],
      'layout':{'title':'Veteran Suicide Methods','annotations':[
            {'font':{'size':20},
            'showarrow':False,'text':'Male','x':0.18,'y':0.5},
            {'font':{'size':20},'showarrow':False,'text':'Female','x':0.85,'y':0.5
            }
        ]}}


py.iplot(fig,filename='Veteran Suicide Methods')



In [59]:
suicide_methods_all = [0.502,1-0.502] # Firearms, others

suicide_methods_veterans = pd.read_excel('Methods.xlsx')
for index, row in suicide_methods_veterans.iterrows():
    row['% by Firearm'] = pd.to_numeric(str(row['% by Firearm']).replace('%',''),errors='coerce')
    row['% by Suffocation'] = pd.to_numeric(str(row['% by Suffocation']).replace('%',''),errors='coerce')
    row['% by Poisoning'] = pd.to_numeric(str(row['% by Poisoning']).replace('%',''),errors='coerce')
suicide_methods_veterans = suicide_methods_veterans.dropna(subset=['% by Firearm'],axis=0,how='any')
suicide_methods_veterans['% by Firearm'] = pd.to_numeric(suicide_methods_veterans['% by Firearm'],errors='coerce')
suicide_methods_veterans = suicide_methods_veterans.rename(columns={'State':'state'})
suicide_methods_veterans.head()


state  % by Firearm  % by Suffocation % by Poisoning  Other  Total  \
0     Alabama         0.808             0.108          0.085    NaN  1.001   
2     Arizona         0.730             0.143           0.14    NaN  1.013   
3    Arkansas         0.720             0.140           0.14    NaN      1   
4  California         0.595             0.187          0.142  0.076      1   
5    Colorado         0.669             0.180          0.152    NaN  1.001   

   % by Firarm All  
0            0.676  
2            0.554  
3            0.627  
4            0.378  
5            0.500

In [24]:
diff_mean = np.mean(suicide_methods_veterans['% by Firearm']-suicide_methods_veterans['% by Firarm All'])
print(diff_mean)
suicide_methods_veterans[['% by Firearm','% by Firarm All']].head()                  

0.14711904761904765


% by Firearm  % by Firarm All
0         0.808            0.676
2         0.730            0.554
3         0.720            0.627
4         0.595            0.378
5         0.669            0.500

The average percentage of veterans using firearm for suicide is 14.7% higher than the percentage for the general population.

In [26]:
suicide_methods_veterans.head()

State  % by Firearm  % by Suffocation % by Poisoning  Other  Total  \
0     Alabama         0.808             0.108          0.085    NaN  1.001   
2     Arizona         0.730             0.143           0.14    NaN  1.013   
3    Arkansas         0.720             0.140           0.14    NaN      1   
4  California         0.595             0.187          0.142  0.076      1   
5    Colorado         0.669             0.180          0.152    NaN  1.001   

   % by Firarm All  
0            0.676  
2            0.554  
3            0.627  
4            0.378  
5            0.500

In [93]:
methods_rural = rate_rural_urban[['state','percent_rural']].merge(suicide_methods_veterans[['state','% by Firearm']],on='state')
methods_rural = methods_rural.merge(characteristics[['state','VA','suicide_rate']],on='state')
methods_rural.corr()

percent_rural  % by Firearm        VA  suicide_rate
percent_rural       1.000000      0.415050  0.341880      0.327591
% by Firearm        0.415050      1.000000  0.429806      0.297092
VA                  0.341880      0.429806  1.000000      0.340881
suicide_rate        0.327591      0.297092  0.340881      1.000000

In [88]:
rural_fire = go.Scatter(x=methods_rural['percent_rural'],
          y = methods_rural['% by Firearm'],
          mode = 'markers',
                       name = 'Rural',
                       text=rate_rural_urban['state'],
                       
          marker = dict(size=10,color='orange',line=dict(width=2,color='black')))

data_area = [rural_fire]

layout = dict(title='Percentage of Veterans in Rural Areas vs. Firearm Use',
             yaxis=dict(title='Firearm as Suicide Method (%)'),
             xaxis = dict(title='Percentage of Veterans in Rural Areas'))
fig = dict(data=data_area,layout=layout)
py.iplot(fig, filename='rural_fire')

In [81]:
methods_rural_income = methods_rural.merge(characteristics[['state','200k+']],on='state')
methods_rural_income.corr()

percent_rural  % by Firearm  suicide_rate     200k+
percent_rural       1.000000      0.415050      0.327591 -0.466306
% by Firearm        0.415050      1.000000      0.297092 -0.383216
suicide_rate        0.327591      0.297092      1.000000 -0.386377
200k+              -0.466306     -0.383216     -0.386377  1.000000

In [102]:
characteristics.corr()['suicide_rate']

total_vet_pop   -0.266475
female           0.153577
med_inc         -0.208567
gulf2            0.148778
gulf1            0.363383
vietnam          0.460681
korean          -0.407222
world2          -0.355355
peace           -0.285262
bachelor        -0.205426
enrolled         0.148345
uninsured        0.572600
disabil          0.357463
VA               0.395031
18-29            0.251435
30-44            0.332506
45-54            0.055993
55-64            0.516001
65+             -0.365545
-10k            -0.057223
10k-14.9k        0.015760
15k-24.9k        0.016018
25k-34.9k        0.144697
35k-49.9k        0.265868
50k-74.9         0.516602
75k-99.9k        0.194078
100k-149.9k     -0.163146
150k-199.9k     -0.276720
200k+           -0.384025
suicide_rate     1.000000
Name: suicide_rate, dtype: float64

In [108]:
ptsd = pd.read_excel('PTSD Data.xlsx',sheet_name = 0)
ptsd = ptsd.rename(columns={'State':'state'})
ptsd = ptsd.merge(rate_rural_urban[['state','percent_rural']])
ptsd.corr()

VISN  \
VISN                                                1.000000   
Total VHA Service Users                             0.087791   
Number of Service Users with \na Confirmed Diag...  0.063974   
Percentage of Users with PTSD                      -0.117438   
Outpatient mental health encounters\n including...  0.004728   
Average Encounters \nper service user with ptsd    -0.158280   
Number of Veterans with a Possible \nDiagnosis ...  0.055582   
\nAverage O/P Stops by Stop Type \n(Among Veter... -0.055351   
Average O/P Stops by Stop Type\n (Among Veteran... -0.112975   
percent_rural                                      -0.074917   

                                                    Total VHA Service Users  \
VISN                                                               0.087791   
Total VHA Service Users                                            1.000000   
Number of Service Users with \na Confirmed Diag...                 0.861015   
Percentage of Users with PTSD                                     -0.064630   
Outpatient mental health encounters\n including...                 0.739126   
Average Encounters \nper service user with ptsd                   -0.224745   
Number of Veterans with a Possible \nDiagnosis ...                 0.858044   
\nAverage O/P Stops by Stop Type \n(Among Veter...                 0.069822   
Average O/P Stops by Stop Type\n (Among Veteran...                -0.208071   
percent_rural                                                     -0.220162   

                                                    Number of Service Users with \na Confirmed Diagnosis of PTSD\n  \
VISN                                                                                         0.063974                
Total VHA Service Users                                                                      0.861015                
Number of Service Users with \na Confirmed Diag...                                           1.000000                
Percentage of Users with PTSD                                                                0.388703                
Outpatient mental health encounters\n including...                                           0.825746                
Average Encounters \nper service user with ptsd                                             -0.243869                
Number of Veterans with a Possible \nDiagnosis ...                                           0.998134                
\nAverage O/P Stops by Stop Type \n(Among Veter...                                           0.051152                
Average O/P Stops by Stop Type\n (Among Veteran...                                          -0.229639                
percent_rural                                                                               -0.203408                

                                                    Percentage of Users with PTSD  \
VISN                                                                    -0.117438   
Total VHA Service Users                                                 -0.064630   
Number of Service Users with \na Confirmed Diag...                       0.388703   
Percentage of Users with PTSD                                            1.000000   
Outpatient mental health encounters\n including...                       0.319693   
Average Encounters \nper service user with ptsd                         -0.005724   
Number of Veterans with a Possible \nDiagnosis ...                       0.394008   
\nAverage O/P Stops by Stop Type \n(Among Veter...                       0.073480   
Average O/P Stops by Stop Type\n (Among Veteran...                      -0.044787   
percent_rural                                                           -0.006767   

                                                    Outpatient mental health encounters\n including psychological testing  \
VISN                                                                                         0.004728                    